lib/resources/docker.rb

```diff
@@ -0,0 +1,341 @@
+      c_images = []
+      raw_images.each_line { |entry|
+        c_images.push(JSON.parse(entry))
+      }
+      c_images
+    end
+
+    # checks if the system is Debian 8+, Ubuntu 16.04+ and RHEL 7+
+    def is_systemd_system
+      return true if inspec.os.redhat? && inspec.os.release.to_i >= 7
+      return true if inspec.os.name == 'ubuntu' && inspec.os.release.to_f >= 16.04
+      return true if inspec.os.name == 'debian' && inspec.os.release.to_i >= 8
+      false
+    end
+
+    # returns parsed params
+    def parse_systemd_values(stdout)
+      SimpleConfig.new(
+        stdout,
+        assignment_re: /^\s*([^=]*?)\s*=\s*(.*?)\s*$/,
+        multiple_values: false,
+      ).params
+    end
+  end
+
+  class DockerContainer < Inspec.resource(1)
+    name 'docker_container'
+    desc ''
+    example "
+      describe docker_container('simple_container') do
+        it { should exist }
+        it { should be_running }
+        its('command') { should eq '/bin/sh' }
+      end
+    "
+
+    # TODO: we need to define if a simple string is a ID or NAME
+    # TODO: handle short ids
+    def initialize(opts = {})
+      # fallback for simple strings
+      if opts.is_a?(String)
+        @opts = { name: opts }
+      else
+        @opts = opts
+      end
+    end
+
+    def exist?
+      !id.nil?
+    end
+
+    def id
+      container_info.ids[0] if container_info.entries.length == 1
+    end
+
+    def running?
+      status.downcase.start_with?('up') if container_info.entries.length == 1
+    end
+
+    def status
+      container_info.status[0] if container_info.entries.length == 1
+    end
+
+    def labels
+      container_info.labels[0] if container_info.entries.length == 1
+    end
+
+    def ports
+      container_info.ports[0] if container_info.entries.length == 1
+    end
+
+    def command
+      container_info.commands[0] if container_info.entries.length == 1
+    end
+
+    def image
+      container_info.images[0] if container_info.entries.length == 1
+    end
+
+    def repo
+      image.split(':')[0] unless image.nil?
+    end
+
+    def tag
+      image.split(':')[1] unless image.nil?
+    end
+
+    def to_s
+      'Docker Container'
+    end
+
+    private
+
+    def container_info
+      return @info if defined?(@info)
+      opts = @opts
+      @info = inspec.docker.containers.where { id == opts[:id] || names == opts[:name] }
+    end
+  end
+
+  class DockerImage < Inspec.resource(1)
+    name 'docker_image'
+    desc ''
+    example "
+      describe docker_image(image: 'busybox:latest') do
+        it { should exist }
+      end
+    "
+
+    # TODO: handle short image ids
+    def initialize(opts = {})
+      @opts = opts
+
+      # do sanitizion of input values
+      @opts = { image: opts } if opts.is_a?(String)
+
+      unless @opts[:image].nil?
+        unless @opts[:image].index(':').nil?
```

Comment:
Would you be OK changing this to an `if !... else` instead of `unless ... else`? I think most Rubyists try to shy away from `unless ... else` flows.